platforms/nuttx/src/px4/common/px4_layer.cmake

```diff
@@ -20,3 +20,5 @@ if (DEFINED PX4_CRYPTO)
 			crypto_backend
 	)
 endif()
+
+target_link_libraries(px4_layer PRIVATE px4_platform)
```

Comment:
This now links both libraries together (which fixes the linkage error) although only board_common.obj is needed from px4_platform. I seem to remember a way to pluck singular objects from a library target, but I cannot remember the syntax (or I might just be wrong).

If you have a suggestion on how to do this better, I'll be more than happy to fix it in a better way